service/storage_proxy.cc

```diff
@@ -6780,7 +6786,7 @@ void storage_proxy::cancel_write_handlers(noncopyable_function<bool(const abstra
             it->timeout_cb();
         }
         ++it;
-        if (need_preempt()) {
+        if (need_preempt() && it != _cancellable_write_handlers_list->end()) {
```

Comment:
Could just reorder the increment.

Reply:
But it current `it` is already canceled why protect it?

Reply:
Right (I think)